# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/InlineTranslator.cpp
# Commit: 8839a3ba6243
# Full Hash: 8839a3ba62436bd0ea7e5bd07e2b63a8d45b8609
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2023-12-09 09:33:28
# Regressor Bug: 1863914
# File Overlap Count: 2
# Description:
#   Bug 1863914: Use multiple shmem buffers for remote canvas recording. r=aosmond
#   
#   This replaces the use of a single large ring buffer.
#   The buffers are still processed in parallel and are recycled to reduce
#   allocation. Events that do not fit in the default sized buffer have a separate
# ==============================================================================

diff -r f088b7a21045 -r 8839a3ba6243 gfx/2d/InlineTranslator.cpp
--- a/gfx/2d/InlineTranslator.cpp	Fri Dec 08 15:56:39 2023 +0000
+++ b/gfx/2d/InlineTranslator.cpp	Fri Dec 08 16:11:27 2023 +0000
@@ -19,26 +19,6 @@
     : mBaseDT(aDT), mFontContext(aFontContext) {}
 
 bool InlineTranslator::TranslateRecording(char* aData, size_t aLen) {
-  // an istream like class for reading from memory
-  struct MemReader {
-    MemReader(char* aData, size_t aLen) : mData(aData), mEnd(aData + aLen) {}
-    void read(char* s, std::streamsize n) {
-      if (n <= (mEnd - mData)) {
-        memcpy(s, mData, n);
-        mData += n;
-      } else {
-        // We've requested more data than is available
-        // set the Reader into an eof state
-        SetIsBad();
-      }
-    }
-    bool eof() { return mData > mEnd; }
-    bool good() { return !eof(); }
-    void SetIsBad() { mData = mEnd + 1; }
-
-    char* mData;
-    char* mEnd;
-  };
   MemReader reader(aData, aLen);
 
   uint32_t magicInt;